pkg/dyninst/decode/decoder.go

```diff
@@ -52,17 +52,37 @@ type snapshotMessage struct {
 	Service  string   `json:"service"`
 	DDSource string   `json:"ddsource"`
 	DDTags   string   `json:"ddtags"`
-	Logger   Logger   `json:"logger"`
-	Debugger Debugger `json:"debugger"`
+	Logger   logger   `json:"logger"`
+	Debugger debugger `json:"debugger"`
 }
 
-type Logger struct {
+func (s *snapshotMessage) init(decoder *Decoder, event output.Event) {
+	s.Debugger.Snapshot = snapshotData{
+		decoder: decoder,
+		event:   event,
+	}
+}
+
+func (s *snapshotMessage) clear() {
+	s.Debugger.Snapshot = snapshotData{}
+}
+
+type logger struct {
```

Comment:
I filed https://datadoghq.atlassian.net/browse/DEBUG-4111 to track collecting the goroutine ID to put here.